### PR TITLE
Backport the Screen Access Manager from Fedora

### PR DESCRIPTION
**Pull request by M4rtinK**
This pull request backports the implementation of the user interaction config file specification:
https://anaconda-installer.readthedocs.io/en/latest/user-interaction-config-file-spec.html

As Gnome (& this also Gnome Initial Setup) developers have suggested some improvements in the spec this PR should be seen as preliminary for now.

### CODE REVIEW

File: pyanaconda/screen_access.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+                            CONFIG_OPTION_PREFIX + option_name, screen_name)
+
+            else:
+                log.warning("Attempt to set option %s as changed for an unknown screen %s.",
+                            CONFIG_OPTION_PREFIX + option_name, screen_name)
+
+    def get_screen_option_changed(self, screen_name, option_name):
+        """Report if the given option on the specified screen has been changed by the user.
+
+        If the screen or option is unknown or if it syntax is incorrect then the option
```
Comment (poncovka) [inline pyanaconda/screen_access.py]: ... its syntax ...

File: tests/pyanaconda_tests/screen_access_manager_test.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
+# vim:set fileencoding=utf-8
+#
+# Copyright (C) 2016  Red Hat, Inc.
```
Comment (poncovka) [inline tests/pyanaconda_tests/screen_access_manager_test.py]: ... 2017 ...

File: anaconda [hunk 3]
```diff
@@ -1221,6 +1221,10 @@ if __name__ == "__main__":
     # In any case do some actions only after NM finishes its connecting.
     threadMgr.add(AnacondaThread(name=constants.THREAD_WAIT_FOR_CONNECTING_NM, target=wait_for_connecting_NM_thread, args=(ksdata,)))
 
+    # initialize the screen access manager before launching the UI
```
Comment (poncovka) [inline anaconda]: In the commit message, there should be ... [0] see docs/... .

**Comment by M4rtinK** [comment]
@poncovka Yeah, I know about the unit test failures & I'm looking into that. :)

File: pyanaconda/screen_access.py [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+
+    def open_config_file(self, config_path=CONFIG_FILE_PATH):
+        """Try to open an existing config file."""
+        with self._lock:
+            if os.path.exists(config_path):
+                log.info("parsing existing user interaction config file in %s", config_path)
+                with open(config_path, "r") as f:
+                    self._config.readfp(f)
+
+    def write_out_config_file(self, config_path=iutil.getSysroot() + CONFIG_FILE_PATH):
```
Comment (M4rtinK) [inline pyanaconda/screen_access.py]: Good catch! That could indeed cause issues as the value returned by getSysroot() can change during the installation.

File: pyanaconda/screen_access.py (hunk 1 above)
Comment (M4rtinK) [inline pyanaconda/screen_access.py]: Fixed. :)

File: tests/pyanaconda_tests/screen_access_manager_test.py (hunk 2 above)
Comment (M4rtinK) [inline tests/pyanaconda_tests/screen_access_manager_test.py]: Fixed.

File: anaconda (hunk 3 above)
Comment (M4rtinK) [inline anaconda]: Fixed. :)

**Comment by M4rtinK** [comment]
And the test failures have been cleared. :)

**Comment by M4rtinK** [comment]
BTW, looks like this Initial Setup PR will also have to be ported to RHEL 7 IS:
https://github.com/rhinstaller/initial-setup/pull/22
Otherwise IS might crash once SAM starts to be used in Anaconda.

**Comment by M4rtinK** [comment]
This is the IS PR needed once SAM lands in Anaconda:
https://github.com/rhinstaller/initial-setup/pull/25

To avoid breakage both Anaconda with SAM and Initial Setup with this PR should be released in lockstep.

**Pull request merged**